src/syscheckd/registry/registry.c

```diff
@@ -509,206 +534,154 @@ void fim_registry_free_entry(fim_entry *entry) {
+    if (configuration == NULL) {
         return;
     }
 
-    fim_registry_calculate_hashes(new, configuration, data_buffer);
-
-    fim_registry_get_checksum_value(new->registry_entry.value);
+    switch (resultType) {
+        case INSERTED:
+            event_data->evt_data->type = FIM_ADD;
+            break;
 
-    /* DEPRECATED CODE
-    saved->registry_entry.value = fim_db_get_registry_data(syscheck.database, new->registry_entry.key->id,
-                                                           new->registry_entry.value->name);
+        case MODIFIED:
 
-    if (configuration->opts & CHECK_SEECHANGES) {
-        diff = fim_registry_value_diff(new->registry_entry.key->path, new->registry_entry.value->name,
-                                       (char *)data_buffer, new->registry_entry.value->type, configuration);
-    }
+            event_data->evt_data->type = FIM_MODIFICATION;
+            break;
 
-    json_event = fim_registry_event(new, saved, configuration, mode,
-                                    saved->registry_entry.value == NULL ? FIM_ADD : FIM_MODIFICATION, NULL, diff);
+        case DELETED:
+            if (configuration->opts & CHECK_SEECHANGES) {
+                fim_diff_process_delete_value(path, name, arch);
+            }
+            event_data->evt_data->type = FIM_DELETE;
+            break;
 
-    os_free(diff);
+        case MAX_ROWS:
+            if (configuration->opts & CHECK_SEECHANGES) {
+                mdebug1("Couldn't insert '%s' entry into DB. The DB is full, please check your configuration.", path);
+                goto end;
+            }
+            break;
+        default:
+            goto end;
+            break;
+    }
 
-    if (json_event) {
-        if (fim_db_insert_registry_data(syscheck.database, new->registry_entry.value, new->registry_entry.key->id,
-                                        saved->registry_entry.value == NULL ? FIM_ADD : FIM_MODIFICATION) != FIMDB_OK) {
-            // Something went wrong or the DB is full, either way we need to stop.
-            mdebug2(FIM_REGISTRY_FAIL_TO_INSERT_VALUE, new->registry_entry.key->arch == ARCH_32BIT ? "[x32]" : "[x64]",
-                    new->registry_entry.key->path, new->registry_entry.value->name);
-            cJSON_Delete(json_event);
-            fim_registry_free_value_data(saved->registry_entry.value);
-            return;
-        }
+    json_event = fim_dbsync_registry_value_json_event(dbsync_event, configuration, FIM_SCHEDULED, event_data->evt_data,
+                                                      NULL, diff, data_type);
 
-        if (_base_line) {
-            send_syscheck_msg(json_event);
-        }
+    if (json_event && _base_line) {
+        send_syscheck_msg(json_event);
+    }
 
+    end:
         cJSON_Delete(json_event);
-    }
-    fim_db_set_registry_data_scanned(syscheck.database, new->registry_entry.value->name, new->registry_entry.key->id);
+        if(diff != NULL){
+            free(diff);
```

Comment:
Here you could use os_free instead free

Reply:
Done in [commit](https://github.com/wazuh/wazuh/pull/11239/commits/11a6215594b89425c4469067a26e23d1a1a7f5df)